File: Solutions/70. Climbing Stairs/70. Climbing Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        def cb(n):
            if n<=1:
                return 1
            return cb(n-1)+cb(n-2)
        return cb(n)
        
class Solution:
    def climbStairs(self, n: int) -> int:
        dp=[0]*(n+1)
        dp[0]=dp[1]=1
        for i in range(2,n+1):
            dp[i]=dp[i-1]+dp[i-2]
        return dp[n]

class Solution:
    def climbStairs(self, n: int) -> int:
        dp=[0]*2
        dp[0]=dp[1]=1
        for i in range(2,n+1):
            dp[i%2]=dp[(i-1)%2]+dp[(i-2)%2]
        return dp[n%2]

File: Solutions/70. Climbing Stairs/70. Climbing Stairs.java
class Solution {
    public int climbStairs(int n) {
        return cb(n);
    }
    private int cb(int n){
        if(n<=1)
            return 1;
        return cb(n-1)+cb(n-2);
    }
}

class Solution {
    public int climbStairs(int n) {
        int[] dp=new int[n+1];
        dp[0]=dp[1]=1;
        for(int i=2;i<=n;++i)
            dp[i]=dp[i-1]+dp[i-2];
        return dp[n];
    }
}

class Solution {
    public int climbStairs(int n) {
        int[] dp=new int[2];
        dp[0]=dp[1]=1;
        for(int i=2;i<=n;++i)
            dp[i%2]=dp[(i-1)%2]+dp[(i-2)%2];
        return dp[n%2];
    }
}